logging_posix.go

```diff
@@ -95,6 +98,83 @@ func (l Log) Debug() (mlog Log) {
 	return
 }
 
+// Custom type to satisfy io.Writer to use as file logging output, handles
+// log file rotation
+type rotateLogWriter struct {
+	sync.Mutex
+	filename string
+	fd       *os.File
+	maxBytes int64 // Maximum size for log file, 0 means no rotation
+}
+
+func (r *rotateLogWriter) new(filename string, maxbytes int64) error {
+	r.filename = filename
+	r.maxBytes = maxbytes
+	return r.initAndCheck()
+}
+
+func (r *rotateLogWriter) Write(output []byte) (int, error) {
+	var err error
+	err = r.initAndCheck()
+	if err != nil {
+		return 0, err
+	}
+	return r.fd.Write(output)
+}
+
+func (r *rotateLogWriter) initAndCheck() (err error) {
+	defer func() {
+		r.Unlock()
+		if e := recover(); e != nil {
+			err = fmt.Errorf("initAndCheck() -> %v", e)
+		}
+	}()
+	r.Lock()
+	// If we haven't initialized yet, open the log file
+	if r.fd == nil {
+		r.fd, err = os.OpenFile(r.filename, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0640)
+		if err != nil {
+			panic(err)
+		}
+	}
+	// If the maximum file size is set to 0 we will never rotate
+	if r.maxBytes == 0 {
+		return nil
+	}
+	// Check the size of the existing log file, and rotate it if required.
+	// We only keep the current log file and one older one.
+	var fi os.FileInfo
+	fi, err = r.fd.Stat()
+	if err != nil {
+		panic(err)
+	}
+	if fi.Size() < r.maxBytes {
+		return
+	}
+	// Rotate the log and reinitialize it
+	// If the old file already exists remove it
+	rotatefile := r.filename + ".1"
+	_, err = os.Stat(rotatefile)
+	if err == nil || !os.IsNotExist(err) {
+		err = os.Remove(rotatefile)
```

Comment:
Another option would be to suffix the file with `date +%Y%m%d%H%M` and keep many versions, but I'm fine with the current approach too as it prevents filing up disks.